improver_tests/utilities/test_field_texture.py

```diff
@@ -0,0 +1,212 @@
+        thresholds=thresholds,
+        spatial_grid="equalarea",
+    )
+    cube = add_coordinate(cube, [0, 1, 2], "realization", dtype=np.int32)
+    return cube
+
+
+# Begin the unit tests.
+
+
+def test_full_process(multi_cloud_cube):
+    """Test the process function with multi realization/threshold input cube."""
+
+    cube = multi_cloud_cube.extract(iris.Constraint(cloud_area_fraction=DIAG_THRESH))
+    cube.remove_coord("cloud_area_fraction")
+    expected_data = np.where(cube.data[0] == 0.0, 1.0, 0.0)
+
+    clumpiness_result = PLUGIN.process(multi_cloud_cube)
+    np.testing.assert_almost_equal(clumpiness_result.data, expected_data, decimal=4)
+
+
+def test__calculate_ratio(multi_cloud_cube):
+    """Test the _calculate_ratio function with single realization of the input cube."""
+
+    cube = multi_cloud_cube.extract(iris.Constraint(cloud_area_fraction=DIAG_THRESH))
+    cube.remove_coord("cloud_area_fraction")
+    expected_data = np.where(cube.data[0] == 0.0, 1.0, 0.3333)
+
+    PLUGIN.cube_name = "cloud_area_fraction"
+    ratio_result = PLUGIN._calculate_ratio(cube[0], NB_RADIUS)
+    np.testing.assert_almost_equal(ratio_result.data, expected_data, decimal=4)
+
+
+def test__calculate_transitions(multi_cloud_cube):
+    """Test the _calculate_transitions function with a numpy array simulating
+       the input cube."""
+
+    cube = multi_cloud_cube.extract(iris.Constraint(cloud_area_fraction=DIAG_THRESH))
+    cube.remove_coord("cloud_area_fraction")
+
+    expected_data = np.zeros((10, 10), dtype=np.float32)
+    expected_data[1:4, 1:4] = np.array(
+        [[2.0, 1.0, 2.0], [1.0, 0.0, 1.0], [2.0, 1.0, 2.0]]
+    )
+
+    transition_result = PLUGIN._calculate_transitions(cube[0].data)
+    np.testing.assert_almost_equal(transition_result, expected_data, decimal=4)
+
+
+def test_process_single_threshold(single_thresh_cloud_cube):
+    """Test the process function with single threshold version of the multi
+       realization input cube."""
+
+    single_thresh_cloud_cube = iris.util.squeeze(single_thresh_cloud_cube)
+    expected_data = np.where(single_thresh_cloud_cube.data[0] == 0.0, 1.0, 0.0)
+
+    clumpiness_result = PLUGIN.process(single_thresh_cloud_cube)
+    np.testing.assert_almost_equal(clumpiness_result.data, expected_data, decimal=4)
+
+
+def test_process_error(multi_cloud_cube):
+    """Test the process function with a non-binary input cube to raise an error."""
+
+    non_binary_cube = np.where(multi_cloud_cube.data == 0.0, 2.1, 0.0)
+
+    with pytest.raises(Exception) as excinfo:
+        PLUGIN.process(non_binary_cube)
+    assert str(excinfo.value) == "Incorrect input. Cube should hold binary data only"
+
+
+def test_process_no_cloud(no_cloud_cube):
+    """Test the FieldTexture plugin with multi realization input cube that has
+       no cloud present in the field."""
+
+    cube = no_cloud_cube.extract(iris.Constraint(cloud_area_fraction=DIAG_THRESH))
+    cube.remove_coord("cloud_area_fraction")
```

Comment:
This is just for the expected data used in the assert command. The actual plugin call is using no_cloud_cube rather than cube so it still has all the coordinates in it.

Reply:
I've kept it with multiple thresholds because I want to simulate what's going on in the plugin so the expected will be the same as the result. Also, I get a mismatch error if I don't do this.